keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/DeliveryConfig.kt

```diff
@@ -9,4 +10,13 @@ data class DeliveryConfig(
   val artifacts: Set<DeliveryArtifact> = emptySet(),
   val environments: Set<Environment> = emptySet(),
   val apiVersion: String = "delivery.config.spinnaker.netflix.com/v1"
-)
+) {
+  val resources: Set<Resource<*>>
+    get() = environments.flatMapTo(mutableSetOf()) { it.resources }
+
+  fun matchingArtifactByReference(reference: String, type: ArtifactType): DeliveryArtifact? =
+    artifacts.find { it.reference == reference && it.type == type }
+
+  fun matchingArtifactByName(name: String, type: ArtifactType): DeliveryArtifact? =
+    artifacts.find { it.name == name && it.type == type }
```

Comment:
These two methods are the key for finding the resource -> artifact association based on information within the delivery config.